src/librustc_resolve/diagnostics.rs

```diff
@@ -497,6 +497,92 @@ impl Bar {
 ```
 "##,
 
+E0408: r##"
+An "or" pattern was used where the variable bindings are not consistently bound
+across patterns.
+
+Example of erroneous code:
+
+```compile_fail
+match x {
+    Some(y) | None => { /* use y */ } // error: variable `y` from pattern #1 is
+                                      // not bound in pattern #2
```

Comment:
"not bound..." should be aligned with "variable `y` from..." instead of "error:".